app/ui/client/views/app/room.js

```diff
@@ -586,6 +597,11 @@ export const dropzoneEvents = {
 
 		e.stopPropagation();
 
+		if (!userCanDrop() || !settings.get('FileUpload_Enabled')) {
```

Comment:
I think here we can add the `!settings.get('FileUpload_Enabled')` to the `userCanDrop` instead of testing both

Reply:
Also, this check can be done earlier in the event, maybe right after removing the `over` class from the element

Reply:
Done

Reply:
same here: `userCanDrop(this._id)`